contracts/helpers/Rebalancer.sol

```diff
@@ -0,0 +1,185 @@
+    ) external view returns (uint256) {
+        return _getGuaranteedAmountOut(tokenIn, tokenOut, amountIn, orders[tokenIn][tokenOut].guaranteedRate);
+    }
+
+    /// @notice Internal version of `_getGuaranteedAmountOut`
+    function _getGuaranteedAmountOut(
+        address tokenIn,
+        address tokenOut,
+        uint256 amountIn,
+        uint256 guaranteedRate
+    ) internal view returns (uint256 amountOut) {
+        return
+            (amountIn * guaranteedRate * (10 ** IERC20Metadata(tokenOut).decimals())) /
+            (1e18 * (10 ** IERC20Metadata(tokenIn).decimals()));
+    }
+
+    /// @notice Computes the additional subsidy amount in agToken that must be added during the process of a swap
+    /// of `amountIn` of `tokenIn` to `tokenOut`
+    function _getSubsidyAmount(
+        address tokenIn,
+        address tokenOut,
+        uint256 amountAgToken,
+        uint256 amountIn
+    ) internal view returns (uint256 subsidy) {
+        Order storage order = orders[tokenIn][tokenOut];
+        uint256 guaranteedAmountOut = _getGuaranteedAmountOut(tokenIn, tokenOut, amountIn, order.guaranteedRate);
+        // Computing the amount of agToken that must be burnt to get the amountOut guaranteed
+        if (guaranteedAmountOut > 0) {
+            uint256 amountAgTokenNeeded = transmuter.quoteOut(guaranteedAmountOut, agToken, tokenOut);
+            // If more agTokens than what has been obtained through the first mint must be burnt to get to the
+            // guaranteed amountOut, we're taking it from the subsidy budget set
+            if (amountAgToken < amountAgTokenNeeded) {
+                subsidy = amountAgTokenNeeded - amountAgToken;
+                // In the case where the subsidy budget is too small, we may not be able to provide the guaranteed
+                // amountOut to the user
+                if (subsidy > order.subsidyBudget) subsidy = order.subsidyBudget;
+            }
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                      GOVERNANCE                                                    
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Lets governance set an order to subsidize rebalances between `tokenIn` and `tokenOut`
+    /// @dev Before calling this function, governance must make sure that there are enough `agToken` idle
+    /// in the contract to sponsor the swaps
+    function setOrder(
+        address tokenIn,
+        address tokenOut,
+        uint256 subsidyBudget,
+        uint256 guaranteedRate
+    ) external onlyGuardian {
+        Order storage order = orders[tokenIn][tokenOut];
+        uint256 newBudget = budget + subsidyBudget - order.subsidyBudget;
+        if (IERC20(agToken).balanceOf(address(this)) < newBudget) revert InvalidParam();
```

Comment:
should we also check that none of the tokens are agToken + that both tokens are whitelisted on the transmuter